ghcloneall.py

```diff
@@ -425,27 +425,38 @@ def list_gists(self, user, pattern=None):
     def list_repos(self, user=None, organization=None, pattern=None,
                    include_archived=False, include_forks=False,
                    include_private=True, include_disabled=True):
+
+        # User repositories default to sort=full_name, org repositories default
+        # to sort=created.  In theory we don't care because we will sort the
+        # list ourselves, but in the future I may want to start cloning in
+        # parallel with the paginated fetching.  This requires the sorting to
+        # happen before pagination, i.e. on the server side, as I want to
+        # process the repositories alphabetically (both for aesthetic reasons,
+        # and in order for --start-from to be useful).
+
         if organization and not user:
             owner = organization
-            list_url = 'https://api.github.com/orgs/{}/repos'.format(owner)
+            list_url = ('https://api.github.com/orgs/{}/repos'
+                        '?sort=full_name').format(
+                            owner)
         elif user and not organization:
             owner = user
-            list_url = 'https://api.github.com/users/{}/repos'.format(owner)
+            if include_private:
+                # users/$name/repos does not include private repos, so
+                # we have to query for the repos owned by the current
+                # user instead.  This only works if the current token
+                # is associated with that user.
+                list_url = ('https://api.github.com/user/repos'
+                            '?affiliation=owner&sort=full_name')
```

Comment:
Uh, just to double-check, this branch is taken only when you provide a GitHub token?

Because by default include_private is True, but github_token is not set.

What does the GitHub API return if you try to GET /user/repos without any token?  Which user does it pick?

I think I may have been too hasty in merging this...

Reply:
> What does the GitHub API return if you try to GET /user/repos without any token? Which user does it pick?

Yeah, you get a 401 Unauthorized.

I'll add a fix.

Reply:
Sorry, yeah, this effectively requires the GitHub token when include_private is true. That's a breaking change in behavior, so I should have highlighted it.

I'd be happy to collaborate on improving that (docs, validation, etc.).